tests/HydratorTest.php

```diff
@@ -582,20 +598,18 @@ public function __construct(
     public function testCountParameterAttributeHandle(): void
     {
         $counterResolver = new CounterResolver();
-        $hydrator = new Hydrator(
-            new SimpleContainer([CounterResolver::class => $counterResolver])
-        );
+        $hydrator = new Hydrator();
 
         $hydrator->create(CounterClass::class);
 
-        $this->assertSame(1, $counterResolver->getCount('a'));
-        $this->assertSame(1, $counterResolver->getCount('b'));
-        $this->assertSame(1, $counterResolver->getCount('c'));
+        $this->assertSame(0, $counterResolver->getCount('a'));
```

Comment:
Why? This tests check that `CounterResolver` call once per attribute.

Reply:
`\Yiisoft\Hydrator\Tests\Support\Attribute\CounterResolver` isn't shared with all of the `\Yiisoft\Hydrator\Tests\Support\Attribute\Counter` attributes in the `\Yiisoft\Hydrator\Tests\Support\Classes\CounterClass` because default hydrator creates a new resolver on each new attribute. If you use di conatiner resolver it will preserve the state inside the resolver.

Reply:
Oh I got it. Reverted.